From a9d62fa7f34465cbf22d6bffa4bc178d0b68248a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Tue, 27 Feb 2024 15:58:08 +0100
Subject: [PATCH 0906/1141] arm64: dts: st: add CRYP2 in stm32mp25 device tree
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Since the stm32mp25xf and stm32mp25xc SoC can have CRYP2 available, it
should be added to the device tree.

Change-Id: Ie228f35525f435b682c752194f8e141c330ff3ba
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/363299
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp25xc.dtsi | 14 ++++++++++++++
 arch/arm64/boot/dts/st/stm32mp25xf.dtsi | 14 ++++++++++++++
 2 files changed, 28 insertions(+)

--- a/arch/arm64/boot/dts/st/stm32mp25xc.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp25xc.dtsi
@@ -19,4 +19,18 @@
 		power-domains = <&CLUSTER_PD>;
 		status = "disabled";
 	};
+
+	cryp2: crypto@42040000 {
+		compatible = "st,stm32mp1-cryp";
+		reg = <0x42040000 0x1000>;
+		interrupts = <GIC_SPI 180 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&rcc CK_BUS_CRYP2>;
+		resets = <&rcc CRYP2_R>;
+		dmas = <&hpdma 140 0x40 0x3021 0x0>,
+		       <&hpdma 141 0x43 0x3012 0x0>;
+		dma-names = "in", "out";
+		feature-domains = <&rifsc STM32MP25_RIFSC_CRYP2_ID>;
+		power-domains = <&CLUSTER_PD>;
+		status = "disabled";
+	};
 };
--- a/arch/arm64/boot/dts/st/stm32mp25xf.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp25xf.dtsi
@@ -19,4 +19,18 @@
 		power-domains = <&CLUSTER_PD>;
 		status = "disabled";
 	};
+
+	cryp2: crypto@42040000 {
+		compatible = "st,stm32mp1-cryp";
+		reg = <0x42040000 0x1000>;
+		interrupts = <GIC_SPI 180 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&rcc CK_BUS_CRYP2>;
+		resets = <&rcc CRYP2_R>;
+		dmas = <&hpdma 140 0x40 0x3021 0x0>,
+		       <&hpdma 141 0x43 0x3012 0x0>;
+		dma-names = "in", "out";
+		feature-domains = <&rifsc STM32MP25_RIFSC_CRYP2_ID>;
+		power-domains = <&CLUSTER_PD>;
+		status = "disabled";
+	};
 };
